hazelcast-cloud/src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -59,6 +66,29 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) {
         attributes.put("X-Amz-Expires", "30");
     }
 
+    private void getKeysFromIamRole() {
+        try {
+            String query = "latest/meta-data/iam/security-credentials/" + awsConfig.getIamRole();
+            URL url = new URL("http", "169.254.169.254", -1, query);

```

Comment:
you can use following constructor here + pull out IP address into a static final.

```    
public URL(String protocol, String host, String file)
            throws MalformedURLException {
        this(protocol, host, -1, file);
    }
```